com.unity.netcode.gameobjects/TestHelpers/Runtime/NetcodeIntegrationTest.cs

```diff
@@ -493,9 +493,17 @@ protected void DestroySceneNetworkObjects()
             var networkObjects = Object.FindObjectsOfType<NetworkObject>();
             foreach (var networkObject in networkObjects)
             {
-                if (CanDestroyNetworkObject(networkObject))
+                if (networkObject != null && CanDestroyNetworkObject(networkObject))
                 {
-                    Object.DestroyImmediate(networkObject);
+                    if (networkObject.gameObject != null)
+                    {
+                        // Destroy the GameObject that holds the NetworkObject component
+                        Object.DestroyImmediate(networkObject.gameObject);
```

Comment:
Why would you not want to also do `Object.DestroyImmediate(networkObject);` here?

Reply:
Because when you destroy a GameObject all attached components and children of the GameObject are destroyed along with it.  The issue was that I was just destroying the NetworkObject and the GameObject (and remaining components) were sticking around.

Reply:
Ah, I see.  Well, let's add comments of "I have an orphaned NO, delete it" and "in deleting the GO all the behaviours are cleaned up also)."

Though I am a bit confused of...
- seems very odd to me that you could have an orphaned NetworkObject.
- ...though is this caused by having `CanDestroyNetworkObject` return true?  Do we like that?  Is it better to just have NO's that don't persist?

Reply:
The list holds references to instantiated NetworkObjects.  I saw a scenario within the TransformInterpolationTests where the NetworkObject was still in the list (i.e. referenced) and the parent GameObject had been despawned/destroyed but the NetworkObject had not.  This might have been maybe due to the reference within NetcodeIntegrationTest?  The TransformInterpolationTests does some parent juggling during its test that could cause this weird scenario where on one frame the parent GameObject and the child GameObject are considered "destroyed" but the test component and associated NetworkObject was not.

Really, this just covers any possible combination of potential integration test weirdness.
I will add more comments behind the reason.